Does not build on FreeBSD

Currently when trying to build Crystal from source on FreeBSD 14.2, getting the following error. 


Running "gmake" >>> 

```
Using /usr/local/bin/llvm-config17 [version=17.0.6]
CRYSTAL_CONFIG_BUILD_COMMIT="85f3cc551" CRYSTAL_CONFIG_PATH='$ORIGIN/../share/crystal/src' SOURCE_DATE_EPOCH="1740594523"  CRYSTAL_CONFIG_LIBRARY_PATH='$ORIGIN/../lib/crystal' ./bin/crystal build -D strict_multi_assign -D preview_overload_order -Dwithout_interpreter  -o .build/crystal src/compiler/crystal.cr -D without_openssl -D without_zlib -D use_pcre2
ld: error: unable to find library -lgc-threaded (this usually means you need to install the development package for libgc-threaded)
cc: error: linker command failed with exit code 1 (use -v to see invocation)
Error: execution of command failed with exit status 1: cc "${@}" -o /home/jsherman/.cache/crystal/home-jsherman-opt-Crystal-crystal-src-ecr-process.cr/macro_run  -rdynamic -L/usr/local/bin/../lib/crystal -lgc-threaded -lpthread -lpthread
gmake: [Makefile:254: .build/crystal] Error 1 (ignored)
```


Currently the version release on FreeBSD ports is far behind, and I am putting in ticket for this as well.

The `-l` is just part of the CLI arguments. I think the wording of the linker error message is a bit misleading. "unable to find library -lgc-threaded" suggests the library is called `-lgc-threaded`, but the name is actually just `gc-threaded`.
On Linux the equivalent error message is a bit clearer as it does not insinuate a library name: `cannot find -lgc-threaded`. 

The package `boehm-gc-threaded` should install the library in `/usr/local/lib/libgc-threaded.so.1`.

--- a/src/gc/boehm.cr
+++ b/src/gc/boehm.cr
@@ -21,7 +21,15 @@ require "crystal/tracing"
 {% end %}
 
 {% if flag?(:freebsd) || flag?(:dragonfly) %}
-  @[Link("gc-threaded")]
+  {% if flag?(:interpreted) %}
+    # FIXME: We're not using the pkg-config name here because that would resolve the
+    # lib flags for libgc including `-lpthread` which the interpreter is not able
+    # to load on systems with modern libc where libpthread is only available as an
+    # (empty) static library.
+    @[Link("gc-threaded")]
+  {% else %}
+    @[Link("gc-threaded", pkg_config: "bdw-gc-threaded")]
+  {% end %}
 {% elsif flag?(:interpreted) %}
   # FIXME: We're not using the pkg-config name here because that would resolve the
   # lib flags for libgc including `-lpthread` which the interpreter is not able

